avocado_vt/plugins/vt_joblock.py

```diff
@@ -0,0 +1,119 @@
+                                           key="dir",
+                                           key_type=str,
+                                           default='/tmp')
+        self.show_lock = settings.get_value(section="plugins.vtjoblock",
+                                            key="show_lock_location",
+                                            key_type=bool,
+                                            default=False)
+        self.lock_file = None
+
+    def _abort(self, message, job):
+        """
+        Aborts the Job by exiting Avocado, adding a failure to the job status
+        """
+        self.log.error(message)
+        sys.exit(exit_codes.AVOCADO_JOB_FAIL | job.exitcode)
+
+    def _set_lock(self, job):
+        if not os.path.isdir(self.lock_dir):
+            msg = ('VT Job lock directory "%s" does not exist... '
+                   'exiting...' % self.lock_dir)
+            self._abort(msg, job)
+
+        pattern = 'avocado-vt-joblock-%(jobid)s-%(uid)s-%(random)s.pid'
+        # the job unique id is already random, but, let's add yet another one
+        rand = ''.join([random.choice(string.ascii_lowercase + string.digits)
+                        for i in xrange(8)])
+        path = pattern % {'jobid': job.unique_id,
+                          'uid': os.getuid(),
+                          'random': rand}
+        path = os.path.join(self.lock_dir, path)
+
+        try:
+            with open(path, 'w') as lockfile:
+                lockfile.write("%u" % os.getpid())
+            self.lock_file = path
+        except IOError as e:
+            msg = ('Failed to create VT Job lock file. Exiting...')
+            self._abort(msg, job)
+
+        if self.show_lock:
+            self.log.info("VT LOCK    : %s", self.lock_file)
+
+    def _get_lock_file(self):
+        if not os.path.isdir(self.lock_dir):
+            return None
+
+        try:
+            files = os.listdir(self.lock_dir)
+            if not files:
+                return None
+        except OSError as e:
+            if e.errno == errno.ENOENT:
+                return None
+
+        pattern = re.compile(r'avocado-vt-joblock-[0-9a-f]{40}-[0-9]+'
+                             '-[0-9a-z]{8}\.pid')
+        for lock_file in files:
+            if pattern.match(lock_file):
+                path = os.path.join(self.lock_dir, lock_file)
+                if os.path.isfile(path):
+                    content = int(open(path, 'r').read())
+                    if int(content) > 0:
+                        return path
+        return None
+
+    def _get_lock_pid(self):
+        lockfile = self._get_lock_file()
+        if lockfile is None:
+            return 0
+        return int(open(lockfile, 'r').read())
+
+    def _lock(self, job):
+        lock_pid = self._get_lock_pid()
+        if lock_pid > 0:
+            msg = ("Avocado-VT job lock acquired by PID %u. "
+                   "Aborting..." % lock_pid)
```

Comment:
It's a nice custom to report whether the PID exists. The 2 extra lines could be very helpful. As before I'd also prefer to list the lockfile as it can help solving the issue with left-over locks (it's simple when one uses just `/tmp` but if people use various different locations, this might become painful).